app/cdap/components/NamespaceAdmin/Preferences.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import { connect } from 'react-redux';
+import SortableTable from 'components/SortableTable';
+import { IPreference } from 'components/NamespaceAdmin/store';
+import Button from '@material-ui/core/Button';
+import SetPreferenceModal, {
+  PREFERENCES_LEVEL,
+} from 'components/FastAction/SetPreferenceAction/SetPreferenceModal';
+import ee from 'event-emitter';
+import globalEvents from 'services/global-events';
+import { getPreferences } from 'components/NamespaceAdmin/store/ActionCreator';
+
+const eventEmitter = ee(ee);
+
+const useStyle = makeStyles((theme) => {
+  return {
+    subtitleSection: {
+      marginBottom: '15px',
+    },
+  };
+});
+
+const PREFERENCES_TABLE_HEADERS = [
+  {
+    property: 'key',
+    label: 'Key',
+    defaultSortBy: true,
+  },
+  {
+    property: 'value',
+    label: 'Value',
+  },
+  {
+    property: 'scope',
+    label: 'Scope',
+  },
+];
+
+interface IPreferencesProps {
+  preferences: IPreference[];
+}
+
+const PreferencesView: React.FC<IPreferencesProps> = ({ preferences }) => {
+  const classes = useStyle();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  useEffect(() => {
+    eventEmitter.on(globalEvents.NSPREFERENCESSAVED, getPreferences);
+
+    return () => {
+      eventEmitter.off(globalEvents.NSPREFERENCESSAVED, getPreferences);
+    };
+  }, []);
+
+  function toggleModal() {
+    setIsModalOpen(!isModalOpen);
+  }
+
+  function renderTableBody(prefs) {
+    return (
+      <tbody>
+        {prefs.map((pref, index) => {
+          return (
+            <tr key={index}>
+              <td title={pref.key}>{pref.key}</td>
+              <td title={pref.value}>{pref.value}</td>
+              <td>{pref.scope}</td>
+            </tr>
+          );
+        })}
+      </tbody>
+    );
+  }
+
+  return (
+    <div>
+      <div className={classes.subtitleSection}>
+        <Button color="primary" variant="contained" onClick={toggleModal}>
+          Edit
+        </Button>
+      </div>
+      <SortableTable
```

Comment:
Can we use the `Table` that we introduced recently for consistency? May be modify `SortableTable` to use the new components, but feel that might be a bigger change.